api/utils/keys/piv_service_fake.go

```diff
@@ -0,0 +1,34 @@
+//go:build pivtest
+
+// Copyright 2025 Gravitational, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package keys
+
+import (
+	"github.com/gravitational/teleport/api/utils/keys/hardwarekey"
+)
+
+// TODO(Joerger): Rather than using a global service, clients should be updated to
+// create a single YubiKeyService and ensure it is reused across the program
+// execution. At this point, it may make more sense to directly inject the mocked
+// hardware key service into the test instead of using the pivtest build tag to do it.
+var mockedHardwareKeyService = hardwarekey.NewMockHardwareKeyService(nil /*prompt*/)
+
+// Returns a globally shared [hardwarekey.MockHardwareKeyService]. Test callers should
+// prefer [hardwarekey.NewMockHardwareKeyService] when possible.
+func NewYubiKeyService(prompt hardwarekey.Prompt) *hardwarekey.MockHardwareKeyService {
```

Comment:
Can `MockHardwareKeyService` live in an internal package? Ideally it wouldn't be exposed in our API module if it's only needed for testing.

Reply:
The problem is that we currently use the `pivtest` build tag in `piv_service_fake.go` to inject this mock service into `api/utils/keys.NewYubiKeyService` for some `tsh` tests. Mo moving the mock service or `piv_service_fake` out of /api would break the dependency chain.

I think the best solution would be to have `tsh` tests inject the mock hardware key service in tests instead of the `pivtest` workaround, but that change would be better suited for a follow up rather than ballooning this PR. I don't think it would even work without the [changes here](https://github.com/gravitational/teleport/pull/53707). WDYT?